File: GreatestCommonDivisor.py
#!/usr/bin/env python3

# If (m > n), then GCD(m, n) = GCD(n, m%n) "Euclid's Algorithm"

def main():
    print(GCD(43, 13, []))
    print(GCD(96, 28, []))

def GCD(m, n, l):
    if (n > m): # Swapping the values if n > m (m is always > n)
        m, n = n, m
    while (m % n): # while (m % n != 0)
        if not m % n == 0:
            val = m % n;
            # Wrong cod is:
            # n = val
            # m = n
            # Python style of solving the problem:
            m, n = n, val
            # Note that: n, m = val, n
    return val

if __name__ == "__main__": main()

File: GreatestCommonDivisor.c
/* GCD of Two +ve Numbers
   Idea: if (m > n) gcd(m, n) = gcd(n, m%n)
       Called the Euclid's algorithm (around 300 B.C.E)
*/
#include <stdio.h>
int GCD(int x, int y);
void main(void) {
    int x, y;
    scanf("%d%d", &x, &y);
    printf("%d\n", GCD(x, y));
}
int GCD(int x, int y) {
    if (x < y) { // Swapping the values (x is always > y)
        int temp = x;
        x = y;
        y = temp;
    }
    int val;
    while (x % y) { // while x % y != 0
        if (! x % y == 0) {
            val = x % y;
            // The wrong code is:
            // y = val;
            // x = y;
            x = y;
            y = val;
        }
        else
            break;
    }
    return val;
}
